frontend/src/utilities/__tests__/router.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { testHook } from '~/__tests__/unit/testUtils/hooks';
+import { setQueryArgument, removeQueryArgument } from '~/utilities/router';
+
+// Mock the NavigateFunction
+const navigate = jest.fn();
+
+describe('setQueryArgument', () => {
+  it('should set a query argument and navigate', () => {
+    // Define the key and value for the query parameter
+    const key = 'param';
+    const value = 'example';
+    // Ensure the query parameter does not exist initially
+    const initialUrl = new URL(window.location.href);
+    initialUrl.searchParams.set(key, value);
+    // Call the hook to set the query parameter
+    testHook(setQueryArgument)(navigate, key, value);
+    // Check if navigate is called with the expected URL
+    const expectedUrl = `${initialUrl.pathname}?${key}=${value}`;
+    expect(navigate).toHaveBeenCalledWith(expectedUrl, { replace: true });
+  });
+  it('should not navigate if query argument is already set to the same value', () => {
+    // Define the key and value for the query parameter
+    const key = 'param';
+    const value = 'example';
+    // Ensure the query parameter is already set to the same value initially
+    const initialUrl = new URL(window.location.href);
+    initialUrl.searchParams.set(key, value);
+
+    // Simulate an existing query parameter with the same value
+    window.history.pushState({}, '', `?${key}=${value}`);
+    // Call the hook to set the query parameter with the same value
+    testHook(setQueryArgument)(navigate, key, value);
+
+    // Check if navigate is not called
+    expect(navigate).not.toHaveBeenCalled();
+  });
+  describe('removeQueryArgument', () => {
+    it('should remove a query argument and navigate', () => {
```

Comment:
This tests relies on the setup of `key` param from the previous test. Tests should be setup to function in isolation.

Reply:
This test _still_ relies on the previous test to set the `key` `param=example`. Run this test in isolation to see the failure.

These lines DO NOT update `window.location` it only updates the `initialUrl` variable which is a _copy_.
```
      const initialUrl = new URL(window.location.href);
      initialUrl.searchParams.set(key, 'example');
```